lib/segment/src/index/field_index/full_text_index/posting_list.rs

```diff
@@ -243,7 +241,10 @@ impl<'a> CompressedPostingVisitor<'a> {
         }
     }
 
-    pub fn contains(&mut self, val: &PointOffsetType) -> bool {
+    // Check if the next value is in the compressed posting list.
+    // This function reuses the decompressed chunk to avoid unnecessary decompression.
+    // It is useful when the visitor is used to check the values in the increasing order.
+    pub fn contains_next(&mut self, val: &PointOffsetType) -> bool {
```

Comment:
Could we extend the comment to explain why `&mut` is needed?

I assume it's there to be careful about something specifically.

Reply:
Commened the same question below. This method is defined in separate help structure which reuse decompressed data to avoid unnecessary decompression. Help structure was commented. Added comments also to this function. And renamed into contains_next

Reply:
`contains_next` still doesn't highlight the need for mutability.